### ISSUE TITLE
Bring Android minsdk down to 16 to support all versions of android Flutter supports

### ISSUE THREAD

**Issue by mitchhymel** [mitchhymel/local_notifications#10]

**Comment by mitchhymel** [comment]
This seems doable by swapping out the Android APIs with their corresponding Compat APIs. 

The one weird thing is that Android 16 and 17 have a bug where the app requires Vibrate permission else an exception will be thrown when calling NotificationManager.notify(). See this [StackOverflow post for reference](https://stackoverflow.com/questions/13602190/java-lang-securityexception-requires-vibrate-permission-on-jelly-bean-4-2).

A workaround to this could be to add the following to the AndroidManifest
```
<uses-permission android:name="android.permission.VIBRATE" android:maxSdkVersion="17"/>
```

**Comment by mitchhymel** [comment]
Completed with commit 12ab6443aa53bbd4fb3f31fb52c3a99e38af523d and included in version 0.0.5

**Comment by aaronfg** [comment]
love the progress you've made with this! Much appreciated

**Comment by huextrat** [comment]
@mitchhymel Apparently the maxSdkVersion must be at least the 18, otherwise the app is not accepted on the PlayStore

**Comment by huextrat** [comment]
@mitchhymel I don't really know but I uploaded it on the PlayStore and I got an alert because the permission for vibration had to be equal 18 at least. 
So I just updated to maxSdk 18 the permission and it works fine, I can't test it atm on low SDK.

I never got this error before because you don't have to set the maxSdkVersion if you use the NotificationBuilder in Android. But I think if we don't set the maxSdk for the vibration it can works. 
It's not a big deal to ask for vibration permission on every phones.
